### PR TITLE
Add predefined attribute section and classes

### PR DESCRIPTION
**Pull request by Girgias**
Rough work but this *should* work.

However I somehow now always hit some memory exhaustion problem when I try to render it although I managed to render it the first time round...

### CODE REVIEW

**Review by TimWolla — COMMENTED** [review]
Something other that comes to my mind when seeing this: The documentation does not appear to support showing applied attributes yet. For the attribute documentation it would be useful to have an indication of the legal attribute targets within class definition.

**Comment by cmb69** [comment]
Thank you for this! I didn't have any rendering issues, even after hooking the new section up in doc-base/manual.xml.in. :)

However, I don't think that language/predefined/attributes/version.xml can work. Instead, this info should be put in language/predefined/versions.xml, and the names of the classes need to be retained (the file names don't matter here).

**Comment by Girgias** [comment]
> Thank you for this! I didn't have any rendering issues, even after hooking the new section up in doc-base/manual.xml.in. :)
> 
> However, I don't think that language/predefined/attributes/version.xml can work. Instead, this info should be put in language/predefined/versions.xml, and the names of the classes need to be retained (the file names don't matter here).

So that's indeed accurate, and I could start rendering again (might have been my RAM being full who knows) but even if PhD stops complaining it still doesn't want to render the version for me >_>

File: language/predefined/attributes/sensitive-parameter.xml [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+ <titleabbrev>SensitiveParameter</titleabbrev>
+
+ <partintro>
+
+  <section xml:id="sensitive-parameter.intro">
+   &reftitle.intro;
+   <para>
+    <!-- TODO Link -->
+    This attribute is used to mark a parameter that is sensitive and should
+    have its value censored if present in a stack trace.
```
[the remark this answers is not in the document]
Comment (Girgias) [reply language/predefined/attributes/sensitive-parameter.xml]: Changed

File: language/predefined/attributes/allow-dynamic-properties.xml [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+<phpdoc:classref xml:id="class.allow-dynamic-properties" xmlns:phpdoc="http://php.net/ns/phpdoc" xmlns="http://docbook.org/ns/docbook" xmlns:xlink="http://www.w3.org/1999/xlink" xmlns:xi="http://www.w3.org/2001/XInclude">
+ <title>The AllowDynamicProperties class</title>
+ <titleabbrev>AllowDynamicProperties</titleabbrev>
+
+ <partintro>
+
+  <section xml:id="allow-dynamic-properties.intro">
+   &reftitle.intro;
+   <para>
+    <!-- TODO Link -->
```
Comment (cmb69) [inline language/predefined/attributes/allow-dynamic-properties.xml:11]: If there is no PR regarding a general description of dynamic properties, I think it's okay to leave this TODO comment in.

File: language/predefined/attributes/allow-dynamic-properties.xml [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+    This attribute is used to mark classes that allow Dynamic properties.
```
Comment (cmb69) [inline language/predefined/attributes/allow-dynamic-properties.xml:12]: nit
```suggestion
    This attribute is used to mark classes that allow dynamic properties.
```

File: language/predefined/attributes/allow-dynamic-properties.xml [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+    <classsynopsisinfo>
+     <ooclass>
+      <modifier>final</modifier>
+      <classname>AllowDynamicProperties</classname>
+     </ooclass>
+    </classsynopsisinfo>
+   </classsynopsis>
+
+  </section>
+
```
Comment (cmb69) [inline language/predefined/attributes/allow-dynamic-properties.xml:34]: Maybe add a small example of the behavior without and with the `AllowDynamicProperties` attribute.

File: language/predefined/attributes/sensitive-parameter.xml [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+<phpdoc:classref xml:id="class.sensitive-parameter" xmlns:phpdoc="http://php.net/ns/phpdoc" xmlns="http://docbook.org/ns/docbook" xmlns:xlink="http://www.w3.org/1999/xlink" xmlns:xi="http://www.w3.org/2001/XInclude">
+ <title>The SensitiveParameter class</title>
+ <titleabbrev>SensitiveParameter</titleabbrev>
+
+ <partintro>
+
+  <section xml:id="sensitive-parameter.intro">
+   &reftitle.intro;
+   <para>
+    <!-- TODO Link -->
```
Comment (cmb69) [inline language/predefined/attributes/sensitive-parameter.xml:11]: I don't think there is a better place to explain this attribute than on this page. Maybe use some word from the [RFC](https://wiki.php.net/rfc/redact_parameters_in_back_traces) (or maybe @TimWolla wants to write something). And we should also add documentation for the `SensitiveParameterValue` class here (separate page).

**Review by cmb69 — COMMENTED** [review]
Could you please also add a page for `ReturnTypeWillChange`?

File: language/predefined/attributes/sensitive-parameter.xml (hunk 5 above)
Comment (Girgias) [reply language/predefined/attributes/sensitive-parameter.xml:11]: I'd prefer letting @TimWolla flesh it out properly and have the current page be like a "stub".

**Review by cmb69 — APPROVED** [review]
Thank you!

Don't forget to hook it in doc-base.

File: language/predefined/attributes/sensitive-parameter.xml (hunk 5 above)
Comment (TimWolla) [reply language/predefined/attributes/sensitive-parameter.xml:11]: @cmb69 The attribute should likely get a mention + example somewhere on the Exception page, for discoverability. That's why I complained about the “Example ordering” in the first place :smiley: Other than that, I can write the documentation for the attribute in a follow-up PR as time permits. For now I'll focus on ext/random as that one is likely less obvious than the SensitiveParameter attribute which was showcased sufficiently in various channels.

**Pull request merged**